### PR TITLE
    deploy: remove external-attacher sidecar from cephfs deployment

### PR DESCRIPTION
**Pull request by humblec**
CephFS CSI driver dont have/advertise controller publish/unpublish capabilities, thus dont need attacher sidecar for its operations.
Also this removal take care of the burden on unncessary operation (round-trip) in the preparation of a volume for a container, and requires CSI Drivers to deploy an unnecessary sidecar container (external-attacher).

Things covered in this PR 
- [x]  Removed attacher sidecar from deploy yamls and also from Helm
- [x] Removed/adjusted RBAC supporting this removal
- [x] CephFS CSI driver object have been changed to `attachRequired:false` param as attachment is no longer required.  

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

### CODE REVIEW

File: deploy/cephfs/kubernetes/csi-cephfsplugin-provisioner.yaml [hunk 1]
```diff
@@ -90,60 +90,6 @@ spec:
...
-            - "--leader-election=true"
-            - "--retry-interval-start=500ms"
-          env:
-            - name: ADDRESS
-              value: /csi/csi-provisioner.sock
-          imagePullPolicy: "IfNotPresent"
-          volumeMounts:
-            - name: socket-dir
-              mountPath: /csi
-        - name: csi-cephfsplugin
```
Comment (Madhu-1) [inline deploy/cephfs/kubernetes/csi-cephfsplugin-provisioner.yaml:107]: Even if its a WIP why cephfsplugin is removed?

**Comment by nixpanic** [comment]
> CephFS CSI driver dont need attacher sidecar for its operations.

It helps if you explain the **why**.

**Comment by humblec** [comment]
> > CephFS CSI driver dont need attacher sidecar for its operations.
> 
> It helps if you explain the **why**.

Sure . Updated the details in the PR description.

**Comment by humblec** [comment]
@Madhu-1 @nixpanic comments are addressed.. ptal.. thanks!

**Comment by Madhu-1** [comment]
As mentioned here https://github.com/ceph/ceph-csi/issues/1106#issuecomment-634424544 This was already attempted a couple of times https://github.com/ceph/ceph-csi/issues/1106#issuecomment-634659533, Are we sure removing is not having any problem. Have we don't all required testing to make sure no problem occurs if we remove it?

**Comment by humblec** [comment]
> As mentioned here [#1106 (comment)](https://github.com/ceph/ceph-csi/issues/1106#issuecomment-634424544) This was already attempted a couple of times [#1106 (comment)](https://github.com/ceph/ceph-csi/issues/1106#issuecomment-634659533), Are we sure removing is not having any problem. Have we don't all required testing to make sure no problem occurs if we remove it?

@Madhu-1 As mentioned in the reply thread https://github.com/ceph/ceph-csi/issues/1106#issuecomment-875737624, it should be fine to remove .. I dont expect any issues due to the same.  More or less, many CSI drivers dont use this sidecar at all as it is unwanted. so we are good :+1:

**Review by Madhu-1 — COMMENTED** [review]
@humblec please squash the commits into one or 2 IMO we don't need to have many commits to for this one. and also retest can be done once PR gets approval.

**Comment by humblec** [comment]
> @humblec please squash the commits into one or 2 IMO we don't need to have many commits to for this one. and also retest can be done once PR gets approval.

@Madhu-1 Done.. ptal.. thanks!

**Comment by humblec** [comment]
@Madhu-1 comments are addresssed. ptal. thanks..

**Review by Madhu-1 — APPROVED** [review]
There was no need to have a different commit for the Doc. ideally all changes should go into 1 or 2 commits.

Considering all the tests done as per https://github.com/ceph/ceph-csi/pull/3149#issuecomment-1143559182 to make sure we don't need an attacher anymore approving the PR.

File: deploy/cephfs/kubernetes/csi-cephfsplugin-provisioner.yaml [hunk 2]
```diff
@@ -90,20 +90,6 @@ spec:
           volumeMounts:
             - name: socket-dir
               mountPath: /csi
-        - name: csi-cephfsplugin-attacher
```
Comment (Madhu-1) [inline deploy/cephfs/kubernetes/csi-cephfsplugin-provisioner.yaml:93]: please remove the RBAC for this one in the deploy/cephfs folder.

**Review by Madhu-1 — CHANGES_REQUESTED** [review]
looks like RBAC for kubernetes templates is not removed yet.

File: deploy/cephfs/kubernetes/csi-cephfsplugin-provisioner.yaml (hunk 2 above)
Comment (humblec) [reply deploy/cephfs/kubernetes/csi-cephfsplugin-provisioner.yaml:93]: @Madhu-1 done. thanks.. ptal.

**Review by Madhu-1 — APPROVED** [review]

**Review by yati1998 — APPROVED** [review]

**Comment by humblec** [comment]
@Mergifyio refresh

**Comment by thrashwerk** [comment]
I'm running a k8s 1.24 cluster and I've got a few static CephFS filesystems for which I have created static PVs and PVCs manually and use the PVCs to mount CephFS to pods.

After updating to 3.7.0 new pods just get stuck with an event `Unable to attach or mount volumes: unmounted volumes=[ceph-pvc-storage1], unattached volumes=[kube-api-access-h2rfp ceph-pvc-storage1]: timed out waiting for the condition`.

I don't get how things are supposed to work now. Why doesn't the CephFS PV get attached?

**Comment by Madhu-1** [comment]
@thrashwerk can you please open a new issue?

**Pull request merged**